src/System.Private.Xml/src/System/Xml/Core/HtmlRawTextWriterGenerator.ttinclude

```diff
@@ -0,0 +1,985 @@
+﻿<#@ template debug="false" hostspecific="false" language="C#" #>
+<#@ import namespace="System" #>
+<#@ include file="RawTextWriter.ttinclude" #>// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// WARNING: This file is generated and should not be modified directly.
+// Instead, modify HtmlRawTextWriterGenerator.ttinclude
+
+using System;
+using System.IO;
+using System.Text;
+using System.Xml;
+using System.Xml.Schema;
+using System.Diagnostics;
+using MS.Internal.Xml;
+
+namespace System.Xml
+{
+    internal class <#= ClassName #> : <#= BaseClassName #>
+    {
+        protected ByteStack elementScope;
+        protected ElementProperties currentElementProperties;
+        private AttributeProperties _currentAttributeProperties;
+
+        private bool _endsWithAmpersand;
+        private byte[] _uriEscapingBuffer;
+
+        private string _mediaType;
+        private bool _doNotEscapeUriAttributes;
+
+        protected static TernaryTreeReadOnly elementPropertySearch;
+        protected static TernaryTreeReadOnly attributePropertySearch;
+
+        private const int StackIncrement = 10;
+
+<# if (WriterType == RawTextWriterType.Encoded) { #>
```

Comment:
The starting construct of T4 also generates a linefeed. In order to have only one linefeed after the `public const`, line 36 could be dropped.

Reply:
Based on the tool's behavior, I think the correct statement would be: the starting construct of T4 generates an extra linefeed, only if the condition is not met. So to achieve the correct one line gap behavior, it would be:

```diff
         private const int StackIncrement = 10;
-
<# if (WriterType == RawTextWriterType.Encoded) { #>
+
         public <#= ClassName #>(TextWriter writer, XmlWriterSettings settings) : base(writer, settings)
        {
            Init(settings);
        }
+
<# } #>
-
         public <#= ClassName #>(Stream stream, XmlWriterSettings settings) : base(stream, settings)
```

both before and end gaps